### PR TITLE
[Core] Adding direct get dof method with position

### PR DESCRIPTION
**Pull request by loumalouomega**
Basically that, this method is more direct and only uses position

### CODE REVIEW

File: kratos/includes/node.h [hunk 1]
```diff
@@ -778,6 +787,27 @@ class Node : public Point,  public IndexedObject, public Flags
         KRATOS_ERROR <<  buffer.str() << std::endl;
     }
 
+    /**
+     * @brief Faster get with very simple and plain check in debug, in release it is direct
+     * @param Position Position of the DoF
+     */
+    inline const DofType& FastGetDof(const IndexType Position) const
```
Comment (RiccardoRossi) [inline kratos/includes/node.h]: this method is very risky in the case you have dofs coming from different modelparts with different dof lists. 
We deliberately avoided to add this, as you have no way (not even in debug) to check that the variable you want is the correct one.

I will block for now ... but let's see if @pooyan-dadvand has a different point of view on this

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]

**Comment by loumalouomega** [comment]
That's why is called FastGet... I will remove it in the other PR.

File: kratos/includes/node.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/node.h]: @RiccardoRossi for example a scheme that uses only one ModelPart as input this is right. Let the developer decide. Here you are increasing computational cost for everyone, for the sake of generality; but then complaining when I add several "ifs" for the sake of generality

**Comment by loumalouomega** [comment]
I removed from the other PR BTW

File: kratos/includes/node.h (hunk 1 above)
Comment (RiccardoRossi) [reply kratos/includes/node.h]: From my point of view, i would define FastGetDof(Variable, Position)

and then just to nothing with the variable. At least you could use it in debug mode to see that everythign is fine. Such solution would have no overhead

**Comment by rubenzorrilla** [comment]
If I'm not wrong, in #3411 we discussed about the inconsistencies of the FastGetSolu... and GetSolu... methods and a consensus was reached about their unification. 

IMO, this kind of main operations must have a unique implementation, even if it introduces an overhead, which importance might or might be not remarkable depending on the problem. 

Having this in mind, if the current implementation is not good enough or introduces too overhead, I'd open an Issue and take profit of the upcoming Workshop to discuss it.

**Comment by loumalouomega** [comment]
> If I'm not wrong, in #3411 we discussed about the inconsistencies of the FastGetSolu... and GetSolu... methods and a consensus was reached about their unification.
> 
> IMO, this kind of main operations must have a unique implementation, even if it introduce an overhead, which importance might or might be not remarkable depending on the problem.
> 
> Having this in mind, if the current implementation is not good enough or introduces too overhead, I'd open an Issue and take profit of the upcoming Workshop to discuss it.

OK, I understand. We can discuss it next week

**Comment by pooyan-dadvand** [comment]
This method just would complicate my implementation of the new NodaDataContainer which also stores Dofs and I'm not sure if we can give this possibility. So I would block it for now...

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
This PR is in conflict with the development of new NodaDataContainer which stores Dofs. Please wait for the other one.

**Comment by philbucher** [comment]
is this still active?

**Comment by loumalouomega** [comment]
> is this still active?

I will solve the conflict, but @pooyan-dadvand ask me to wait for a PR I think is already merged

**Comment by RiccardoRossi** [comment]
> If I'm not wrong, in #3411 we discussed about the inconsistencies of the FastGetSolu... and GetSolu... methods and a consensus was reached about their unification.
> 
> IMO, this kind of main operations must have a unique implementation, even if it introduces an overhead, which importance might or might be not remarkable depending on the problem.
> 
> Having this in mind, if the current implementation is not good enough or introduces too overhead, I'd open an Issue and take profit of the upcoming Workshop to discuss it.

@rubenzorrilla just to tell that i don't think we have any consensus on Fast vs Get ...

**Comment by RiccardoRossi** [comment]
also if we go this way we should print a meaningful message like "amesos was disabled, please enable it in order to use this solver"

**Comment by RiccardoRossi** [comment]
can we close this?

**Comment by loumalouomega** [comment]
> can we close this?

Why, I will fix the compilation

**Review by roigcarlo — CHANGES_REQUESTED** [review]
@KratosMultiphysics/technical-committee Does not accept this PR:
The reason is that position is not deterministically going to give a better performance, but a hint which in most cases will work. 
Assumin this FastGetDof can crash in release, and the only way to prevent is change its behaviour to do the same as GetDof.

@KratosMultiphysics/technical-committee also agrees on the need of having a pGetDof with position, which can be implemented in a separate PR.

**Comment by philbucher** [comment]
why is this still open?

**Comment by loumalouomega** [comment]
> 
> 
> why is this still open?

Because I forgot

**Comment by philbucher** [comment]
can we delete the branch?

**Comment by loumalouomega** [comment]
> 
> 
> can we delete the branch?

I need to copy some stuff

**Comment by loumalouomega** [comment]
Now

**Pull request closed without merging**